BootloaderCommonPkg/Library/ShellLib/CmdMtrr.c

```diff
@@ -83,9 +83,21 @@ ShellCommandMtrrFunc (
   IN CHAR16 *Argv[]
   )
 {
-  UINT32  Index, BaseMsr, MaskMsr, Type;
-  BOOLEAN MtrrEnabled, FixedMtrrEnabled, Valid, Verbose;
-  UINT64  DefType, BaseVal, Base, MaskVal, Mask, Range, Limit;
+  UINT32      Index;
+  UINT32      BaseMsr;
+  UINT32      MaskMsr;
+  UINT32      Type;
+  BOOLEAN     MtrrEnabled;
+  BOOLEAN     FixedMtrrEnabled;
+  BOOLEAN     Valid;
+  UINT64      DefType;
+  UINT64      BaseVal;
+  UINT64      Base;
+  UINT64      MaskVal;
+  UINT64      Mask;
+  UINT64      Range;
+  UINT64      Limit;
+  BOOLEAN     Verbose;
 
   Verbose = FALSE;
 

```

Comment:
Please use PrintMtrr() here.

Reply:
the shell command should not use PrintMtrr() since it uses DEBUG to output MTRR message. It means there is output in release build for PrintMtrr(). The shell command should always works regardless debug or release build.

Reply:
The PrintMtrr and CmdMtrr is almost identical except of printing method. Can you make it a little more common? I think MtrrLib needs to have more primitive APIs.

Reply:
do you have any suggestions on how to make it more common?

Reply:
Let's make primitive APIs later when we really need external APIs.
For PrintMtrr(), I think we can add a output_function parameter in PrintMtrr() like ListDir().
if output_function is NULL, we can use DEBUG(). Otherwise, re-route to output_function.

Reply:
I don't get the exact idea on how to do this. how about check this patch firstly, then you help impl this @myguitar ?

Reply:
Sure. Let's merge this first. I will make a follow-up patch.